### PR TITLE
Automatically create items.json using `npm run objgen`

### PR DESCRIPTION
**Pull request by RodEsp**
This allows us to create the `src/items.json` file with a JS script that we can run with `npm`.
We can later incorporate this into a CI/CD flow.

### CODE REVIEW

File: scripts/buildItemsJSON.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+const itemsJSON = JSON.parse(ObjGen.xJson(itemObjGen, {numSpaces: 2}));
+
+// Sort Alphabetically by item.name
+itemsJSON.sort((itemA, itemB) => itemA.name.localeCompare(itemB.name));
+
+// Sort Alphabetically by item.category
+itemsJSON.sort((itemA, itemB) => itemA.category.localeCompare(itemB.category));
+
+fs.writeFileSync('./src/items.json', JSON.stringify(itemsJSON, null, '\t'), 'utf8');
```
Comment (kyleshrader) [inline scripts/buildItemsJSON.js:14]: **Warning, holy war incoming...**

Can you make it two spaces instead of tab-space?

Comment (RodEsp) [reply scripts/buildItemsJSON.js:14]: I used tabs because that's what it was already using and I didn't want to cause a diff of everything.

**Comment by RodEsp** [comment]
Hmm... _everything_ changed around?
There were several items that changed positions because their names changed and therefore their alphabetical ordering changed too (mostly the concrete -> cement stuff) but it shouldn't be _everything_.

**Comment by RodEsp** [comment]
Ok well now after changing to spaces then yes, everything has changed haha.
But it shouldn't matter going forward. The diff won't be huge it'll only show what changed.
image

**Comment by kyleshrader** [comment]
> Hmm... _everything_ changed around?
> There were several items that changed positions because their names changed and therefore their alphabetical ordering changed too (mostly the concrete -> cement stuff) but it shouldn't be _everything_.

Oh gotcha. This has the data from the other PRs will go through it :)

**Comment by kyleshrader** [comment]
!!! Not close and comment! why would they put that right next to comment.

**Pull request merged**